### PR TITLE
update bclconvert module, as used in nf-core/demultiplex

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist
This PR updates to module with modifications needed for the nf-core/demultiplex workflow.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/bclconvert/meta.yml [hunk 1]
```diff
@@ -12,6 +12,10 @@ tools:
       licence: "ILLUMINA"
 
 input:
+  - meta:
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
   - samplesheet:
       type: file
       description: "Input samplesheet"
```
Comment (maxulysse) [inline modules/bclconvert/meta.yml:21]: missing input `run_dir`

File: modules/bclconvert/main.nf [hunk 2]
```diff
@@ -8,15 +8,14 @@ process BCLCONVERT {
...
-    path "*.fastq.gz"               ,emit: fastq
-    path "Reports/*.{csv,xml,bin}"  ,emit: reports
-    path "Logs/*.{log,txt}"         ,emit: logs
-    path "InterOp/*.bin"            ,emit: interop
-    path "versions.yml"             ,emit: versions
+    tuple val(meta), path("**.fastq.gz")            ,emit: fastq
+    tuple val(meta), path("Reports/*.{csv,xml}")    ,emit: reports
+    tuple val(meta), path("Logs/*.{log,txt}")       ,emit: logs
+    tuple val(meta), path("**.bin")                 ,emit: interop
+    tuple val(meta), path("versions.yml")           ,emit: versions
```
[the remark this answers is not in the document]
Comment (matthdsm) [reply modules/bclconvert/main.nf]: fixed

[the remark this answers is not in the document]
Comment (matthdsm) [reply modules/bclconvert/main.nf]: This was the issue that was giving us flak in the other demultiplex workflow PR

**Comment by matthdsm** [comment]
So the main issue here for the test doesn't have anything to do with `bclconvert`, but with the `untar` module. Apparently `untar` expects the ouput directory to have the same name as the tar archive.

**Comment by matthdsm** [comment]
@Emiller88 , I think this one's good to go!

**Comment by Emiller88** [comment]
Is there a reason the docker/singularity tests are still failing?

**Comment by matthdsm** [comment]
They shouldn't... I'll give it a once over. Probably something to do with my last commit

**Comment by matthdsm** [comment]
Fixed it! Small typo

**Comment by matthdsm** [comment]
I think this is as complete as it's going to get. 
Can I get a ✅  and merge?

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**